crates/model/src/order.rs

```diff
@@ -391,6 +326,61 @@ impl OrderCreation {
         });
         signing::keccak256(&hash_data)
     }
+
+    pub fn token_pair(&self) -> Option<TokenPair> {
+        TokenPair::new(self.buy_token, self.sell_token)
+    }
+
+    pub fn uid(&self, domain: &DomainSeparator, owner: &H160) -> OrderUid {
+        let mut uid = OrderUid([0u8; 56]);
+        uid.0[0..32].copy_from_slice(&super::signature::hashed_eip712_message(
+            domain,
+            &self.hash_struct(),
+        ));
+        uid.0[32..52].copy_from_slice(owner.as_fixed_bytes());
+        uid.0[52..56].copy_from_slice(&self.valid_to.to_be_bytes());
+        uid
+    }
+}
+
+// An order as provided to the orderbook by the frontend.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OrderCreation {
+    #[serde(flatten)]
+    pub data: OrderData,
+    pub from: Option<H160>,
+    #[serde(flatten)]
+    pub signature: Signature,
+    pub quote_id: Option<QuoteId>,
+}
+
+impl OrderCreation {
+    /// Recovers the owner address for the specified domain, and then verifies
+    /// it matches the expected address.
+    ///
+    /// Returns the recovered address on success, or an error if there is an
+    /// issue performing the EC-recover or the recovered address does not match
+    /// the expected one.
+    pub fn verify_owner(&self, domain: &DomainSeparator) -> Result<H160, VerificationError> {
+        self.signature
+            .verify_owner(self.from, domain, &self.data.hash_struct())
+    }
+}
+
+impl Default for OrderCreation {
+    // Custom implementation to make sure the default order creation is valid.
+    fn default() -> Self {
+        Self {
+            data: OrderData {
+                valid_to: u32::MAX,
```

Comment:
Can this super long `valid_to` be a problem after https://github.com/cowprotocol/services/pull/224?

Reply:
I just kept this from the existing `impl Default for OrderCreation` (didn't want to change logic too much, and removing it cause quite a few tests to start failing):

https://github.com/cowprotocol/services/blob/becd679bae4276ad514b834afbfe64c2b561c804/crates/model/src/order.rs#L305

Reply:
Ahh, right. I overlooked that `u32::MAX` was already used before. 👌